src/services/RewardService.ts

```diff
@@ -39,6 +39,12 @@ export default class RewardService {
             return { error: 'This reward already disabled by it owner' };
         }
 
+        // Can not claim if reward has an unlockDate and the Now is not greather than unlockDate
```

Comment:
@valeriagrazzini User **should be able to claim** prior to this unlockDate, but **should not be able to withdraw** prior to the unlockDate. 

In your current approach, the smart contract logic becomes dead code since it will never be touched. Move this check to this controller instead: 
```
src/controllers/withdrawals/postPollFinalize.action.ts
```